plugin/src/main/java/org/opensearch/ml/action/agents/TransportRegisterAgentAction.java

```diff
@@ -60,10 +66,42 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<MLRegi
 
     private void registerAgent(MLAgent agent, ActionListener<MLRegisterAgentResponse> listener) {
         Instant now = Instant.now();
-        MLAgent mlAgent = agent.toBuilder().createdTime(now).lastUpdateTime(now).build();
+        String llmId;
+        if (agent.getType().equalsIgnoreCase(MLAgentType.CONVERSATIONAL.name()) && agent.getLlm() != null) {
```

Comment:
If we add a new agent type which also needs LLM, we need to remember to update this part. Not so easy to maintain.

How about change to 
```suggestion
        if (agent.getLlm() != null) {
```

Reply:
done